app/controllers/api/documents_controller.rb

```diff
@@ -8,18 +11,43 @@ class DocumentsController < APIController
     param :keyword, String, required: false
     param :blocks, Array, required: false
     def index
-      documents = DocumentProvider.new(
-        current_site, 
-        params[:document_type], 
-        params[:keyword], 
-        params[:blocks]
-      ).retrieve
-
       if documents
-        render json: documents, meta: { results: documents.size }, root: 'documents'
+        render json: paginated_documents, meta: meta_data, root: 'documents'
       else
         render json: { message: 'Bad request' }, status: 400
       end
     end
+
+    private
+
+    def documents
+      @documents ||= DocumentProvider.new(
+        current_site,
+        params[:document_type],
+        params[:keyword],
+        params[:blocks]
+      ).retrieve
+    end
+
+    def paginated_documents
+      documents.page(page).per(per_page)
+    end
+
+    def meta_data
+      {
+        results: documents.size,
+        page: page,
+        per_page: per_page,
+        total_pages: paginated_documents.total_pages
+      }
+    end
+
+    def page
```

Comment:
I really understand the point that you are seeing but I think it will be more consistent if is the same name of the params that we send. Like "Give me the page 3" ... "Here is your page 3".